float32 thateFL
float32 thetaFR
float32 thetaRL
float32 thetaRR
